### PR TITLE
Fix batch normalization to be able to deal with train() and eval() mo…

### PR DESCRIPTION
**Pull request by dlibenzi**
…des.

### CODE REVIEW

File: torch_xla/csrc/ops/moving_average.h [hunk 1]
```diff
@@ -6,24 +6,20 @@ namespace torch_xla {
 namespace ir {
 namespace ops {
 
-class BatchNormForward : public Node {
+class MovingAverage : public Node {
```
Comment (asuhan) [inline torch_xla/csrc/ops/moving_average.h]: This isn't really a moving average, it's just a linear interpolation. Maybe rename it?

**Comment by dlibenzi** [comment]
> I believe this fixes #220, right?

Also, yes. I had not noticed that bug.
This born from an investigation Daniel and Taylan did on BN.

> How widespread is the use of running mean and variance with training, iow does this change behavior for our ResNet50 benchmark?

I have not measured resnet50 yet, but this is something we cannot leave as is.
BN uses only the mean/variance of the current mini-batch otherwise, and worse, if user code use .train() and .eval() (which they do), accuracy drops massively (MNIST from 1st EPOCH 96% to 33%).

File: torch_xla/csrc/ops/moving_average.h (hunk 1 above)
Comment (dlibenzi) [reply torch_xla/csrc/ops/moving_average.h]: This is used as moving average from BN but I agree, we can have to low level op be named something more generic.
